source/_posts/2020-03-14-2020-using-vyper-with-embark.md

```diff
@@ -0,0 +1,204 @@
+```
+yarn global add embark
+# OR
+npm i -g embark
+```
+
+The rest of this article will assume you have Embark installed globally, and therefore available from the CLI.
+
+### Install Vyper
+If you haven't already done so, make sure to [install Vyper](https://vyper.readthedocs.io/en/latest/installing-vyper.html). As the documentation recommends, be sure to [use a Python virtual environment when installing Vyper](https://vyper.readthedocs.io/en/latest/installing-vyper.html#creating-a-virtual-environment), as it will keep your system-wide packages from being polluted.
+
+## Step 2. Create a &ETH;App
+For this article, we will be creating a demo &ETH;App to use as a base for creating our first Vyper contract. However, if you already have a &ETH;App that you'd like to add Vyper contracts with, simply skip this step.
+
+Creating an Embark demo is easy, simply run the following commands:
+```
+embark demo
+cd embark_demo
+```
+
+## Step 3. Install the Embark Vyper plugin
+Installing the [Embark Vyper](https://github.com/embarklabs/embark/blob/master/packages/plugins/vyper/README.md) plugin in our &ETH;App is extremely simple:
+1. Add the `embark-vyper` package to your &ETH;App:
+```
+yarn add embark-vyper
+# OR
+npm i embark-vyper
+```
+2. Add the `embark-vyper` plugin to `embark.json`:
+```
+// embark.json
+
+// ...
+"plugins": {
+  "embark-ipfs": {},
+  "embark-swarm": {},
+  "embark-whisper-geth": {},
+  "embark-geth": {},
+  "embark-parity": {},
+  "embark-profiler": {},
+  "embark-graph": {},
+  "embark-vyper": {} // <====== add this!
+},
+// ...
+```
+
+## Step 4. Add a Vyper contract to your &ETH;App
+First, let's delete the `SimpleStorage` Solidity contract that comes with the Embark Demo by default:
+```
+rm -rf contracts/simple_storage.sol
+```
+Next, let's create a Vyper contract in the `contracts` folder of our &ETH;App, ie `contract/SimpleStorage.vy` (case is important):
+```
+# contracts/SimpleStorage.vy
+
+storedData: public(int128)
+
+@public
+def __init__(_x: int128):
+  self.storedData = _x
+
+@public
+def set(_x: int128):
+  self.storedData = _x
+```
+The function of this contract is simple: upon creation (deployment), it will store the value we initially give it during deployment. We also have access to a `set()` that will allow to set the value stored in the contract.
+
+### Changing the contract filename
+Because Vyper's constructor is called `__init__`, Embark must take the contract class name from the file name. If you'd prefer to change the name of your contract file to something else like `contracts/simple_storage.vy`, you'll need to update `config/contract.js` to allow Embark to match a contract configuration to a contract file:
+```
+// config/contracts.js
+
+module.exports = {
+  default: {
+    // ...
+    deploy: {
+      simple_storage: {
```

Comment:
Probably worth it to add a comment next to this line saying "Only change this if you changed the file name. I say that, because some people don't read the text and just copy the code snippets and in this case, it's an optional step.